### PR TITLE
Add missing onClick property on on <Navbar.Toggle /> props

### PR DESCRIPTION
**Pull request by LKay**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.
If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by aozgaa** [comment]
pinging @walkerburgin @vsiao @danilojrr @Batbold-Gansukh @octatone. Can you also provide a link to documentation for why the change is needed?

File: react-bootstrap/index.d.ts [hunk 1]
```diff
@@ -528,6 +528,7 @@ declare namespace ReactBootstrap {
 
     // <Navbar.Toggle />
     interface NavbarToggleProps {
+        onClick?: React.MouseEventHandler<{}>;
```
Comment (vsiao) [inline react-bootstrap/index.d.ts]: judging by https://github.com/react-bootstrap/react-bootstrap/blob/master/src/NavbarToggle.js, this should be `React.MouseEventHandler<HTMLButtonElement>`.

Comment (LKay) [inline react-bootstrap/index.d.ts]: Handlers signatures for other components don't specify element types but sure, I can make it more strict.

**Comment by zhengbli** [comment]
@vsiao approve?

**Comment by vsiao** [comment]
👍

**Pull request merged**